respond.src.js

```diff
@@ -267,7 +267,11 @@ window.matchMedia = window.matchMedia || (function( doc, undefined ) {
 			}
 			
 			//inject active styles, grouped by media type
+			var keys = [];
 			for( var k in styleBlocks ){
+				keys.push( k );
```

Comment:
This assumes that the order of iteration for an object is first-in-first-out, e.g. for `styleBlocks.foo = 'foo'; styleBlocks.bar = 'bar'`, foo will come first. Although this is the common implementation in modern browsers, unfortunately you cannot and should not rely on it. It's not defined in the ECMAScript specification. See http://stackoverflow.com/a/280861/3150057.

So for this to work properly, `keys` must be appended-to at the same time as `styleBlocks` is. Then the order will be correct in the `styleBlocks` array. This will also reduce the number of loops occurring.